src/main/java/racing/Racing.java

```diff
@@ -1,35 +1,55 @@
 package racing;
 
+import java.util.Arrays;
+import java.util.Comparator;
 import java.util.List;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
 public class Racing {
 
+    private static final int MIN_ROUND = 1;
+    private static final int MAX_ROUND = 10;
+
     private int totalRound;
     private int nowRound;
     private List<Car> cars;
-    private RandomNumberGenerator numberGenerator;
-
+    private NumberGenerator numberGenerator;
 
-    public Racing(int carCount, int moveCount) {
+    public Racing(String names, int moveCount, NumberGenerator numberGenerator) {
+        validateRound(moveCount);
         this.totalRound = moveCount;
         this.nowRound = 0;
-        this.cars = IntStream.range(0, carCount).mapToObj(a -> new Car()).collect(Collectors.toList());
-        this.numberGenerator = new RandomNumberGenerator();
+        this.cars = Arrays.stream(names.split(",")).map(Car::new).collect(Collectors.toList());
+        this.numberGenerator = numberGenerator;
+    }
+
+    private void validateRound(int moveCount) {
+        if (moveCount < MIN_ROUND || moveCount > MAX_ROUND) {
+            throw new IllegalArgumentException("이동 횟수는 10이하의 자연수만 가능합니다.");
```

Comment:
```suggestion
            throw new IllegalArgumentException("이동 횟수는 " + MAX_ROUND + "이하의 자연수만 가능합니다.");
```

와 같이 작성하면 값이 변경 됐을 떄 메시지를 수정하지 않아도 되겠네요 :)